sklearn/compose/_column_transformer.py

```diff
@@ -246,13 +247,38 @@ def _iter(self, fitted=False, replace_strings=False):
                 elif _is_empty_column_selection(column):
                     continue
 
-            yield (name, trans, column, get_weight(name))
+            yield (name, column, trans, get_weight(name))
+
+    def _validate_tuple_order(self):
+        # check if any given tuple follows the
+        # (name, transformer, column) order.
+        # flip all tuples if that's the case.
+        # remove in v0.22
+        warn_message = ('ColumnTransformer transformer tuples should be '
+                        '(name, column, transformer), whereas '
+                        '(name, transformer, column) was passed; '
+                        'its support is deprecated in 0.20.1 and will be '
+                        'removed in version 0.22. Assuming the deprecated '
```

Comment:
I think the "Assuming the deprecated order for all given tuples" could be a bit confusing right after the deprecation sentence. If someone reads a ``,`` instead of a ``.`` after 0.22 the meaning is completely changed.

Reply:
Does it look better now @amueller ?